lib/velum/kubernetes.rb

```diff
@@ -1,9 +1,36 @@
 # frozen_string_literal: true
+
+require "velum/salt"
+
 module Velum
   # Kubernetes deals with the Kubernetes integration of this application.
   class Kubernetes
     attr_reader :client
 
+    KubeConfig = Struct.new :host, :ca_crt, :client_crt, :client_key
+
+    # Returns the Kubernetes apiserver configuration. It returns a KubeConfig struct.
+    def self.kubeconfig
+      _, ca = Velum::Salt.call action:  "mine.get",
+                               targets: "ca",
+                               arg:     ["ca", "x509.get_pem_entries"]
+      _, apiserver_crt = Velum::Salt.call action:      "cmd.run",
+                                          targets:     "roles:kube-master",
+                                          target_type: "grain",
+                                          arg:         ["cat /etc/pki/minion.crt"]
+      _, apiserver_key = Velum::Salt.call action:      "cmd.run",
+                                          targets:     "roles:kube-master",
+                                          target_type: "grain",
+                                          arg:         ["cat /etc/pki/minion.key"]
+      host = Socket.gethostname
```

Comment:
That's what I thought. So the host should be the ip/hostname of the minion with the k8s-master role. I'm tried to change to this up manually but it doesn't seem to work:

```
The connection to the server 10.17.3.111:443 was refused - did you specify the right host or port?
```
Can you give a try too? Maybe I'm doing something wrong.